### PR TITLE
Documentation update and re-organisation

### PR DESCRIPTION
**Pull request by lildude**
This PR updates and re-organises the documentation.

It makes the README more readable by moving most of the content to individual files under `docs/` and updates the content to be more consistent across all files and reflect the behaviour and requirements once https://github.com/github/backup-utils/pull/375 has been merged.

Most of the overall wording is the same.

/cc @github/backup-utils

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Getting started
+
+ 1. [Download the latest release version][1] and extract the repository using `tar`:
+
+    `tar -xzvf /path/to/github-backup-utils-vMAJOR.MINOR.PATCH.tar.gz`
```
Comment (snh) [inline docs/getting-started.md]: While it is mentioned in the requirements section, additional clarity around which version of the backup utilities to use if you are running GHE 2.10 or earlier would be valuable here, in particular if cloning the repository using `git`.

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    *or* clone the repository using Git:
+
+    `git clone -b stable https://github.com/github/backup-utils.git`
+
+    **Note**: you will need to use [backup-utils v2.11.x][2] or the `legacy` branch to
+    backup and restore GitHub Enterprise 2.10 and earlier.
```
Comment (lildude) [inline docs/getting-started.md]: @snh like this? ☝️

File: README.md [hunk 3]
```diff
@@ -1,24 +1,13 @@
...
-  - **[GitHub Enterprise version requirements](#github-enterprise-version-requirements)**
-- **[Getting started](#getting-started)**
-- **[Migrating from GitHub Enterprise v11.10.34x to v2.0](#migrating-from-github-enterprise-v111034x-to-v20-or-v21)**
-- **[Using the backup and restore commands](#using-the-backup-and-restore-commands)**
-- **[Scheduling backups](#scheduling-backups)**
-- **[Backup snapshot file structure](#backup-snapshot-file-structure)**
-- **[How does backup utilities differ from a High Availability replica?](#how-does-backup-utilities-differ-from-a-high-availability-replica)**
-- **[Support](#support)**
+**Note**: the [GitHub Enterprise version requirements](docs/requirements.md#github-enterprise-version-requirements) have
+changed starting with Backup Utilities v2.13.0.
```
Comment (rubiojr) [inline README.md]: Should we add a date here so folks have a better sense of when this requirement changed?

Comment (lildude) [reply README.md]: Good idea (for those who follow master so don't get versions).

**Pull request merged**